hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetBeanQuery.java

```diff
@@ -169,14 +172,23 @@ private Boolean isTagSelected() {
         return true;
     }
 
+    private Boolean isOverdueFilterEnabled() {
+        if (Boolean.TRUE.equals(overdueState)) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
     @Override
     protected void saveBeans(final List<ProxyTarget> addedTargets, final List<ProxyTarget> modifiedTargets,
             final List<ProxyTarget> removedTargets) {
         // CRUD operations on Target will be done through repository methods
     }
 
     private Boolean anyFilterSelected() {
-        if (status == null && distributionId == null && Strings.isNullOrEmpty(searchText) && !isTagSelected()) {
+        int enabledFiltersCount = Booleans.countTrue(status != null, distributionId != null,
```

Comment:
Don't use `Booleans.countTrue(...)` to hide a `and` conjunction of your condition.